### DISCUSSION TITLE
Contribution graph does not show my commits in my own public repo

### DISCUSSION THREAD

**Discussion by henriquebaron** [community/community#36004] [Other Feature Feedback, Questions, & Ideas]
Commit's that I've done from my work laptop in one of my personal public repos do not show in the contribution graph.

I took the rules into consideration:
- The commits are on the `main` branch.
- The commits are older than 24 hours.
- The commits were signed with the e-mail of my GitHub account. My avatar shows up correctly in the commit list, for example.

For example, in September 27th I made 6 commits in the [angular-course](https://github.com/henriquebaron/angular-course) repository, but the graph shows only one. There are other days in which absolutely no commit is shown.
![image](https://user-images.githubusercontent.com/37144857/195554970-5646a181-7393-4a31-9c8a-078201a7d8c0.png)
![image](https://user-images.githubusercontent.com/37144857/195555216-6ac7bdc9-23ac-4a22-a53c-77a296285c10.png)

What could be the problem?

**Comment by backtoaldo** [comment]
As you specified that you made the commits following the github rules and following the documentation: [Why are my contributions not showing up on my profile?](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile). 

This problem could be caused by different problems

- [x] The first one you can test on your machine is: inside the repository you are working, check your git config user.email - check if the email is the same as your account. Usually for work or testing purposes, you may have it changed. 

Trying to explain the problem:

Sometimes, depending on your email, you may have forgotten some "." or some information in it, your --global email may be correct but your --local or --system email may contain some error.

`git config --global user.email`
`git config --system user.email`
`git config --local user.email`

You can try to remove everything and add it again to your local repository and see if the changes come through, sometimes for n reasons some bug happens.

-----------------------------------------------------------------------------------------------------------------------

- [x] The other way I know needs a few steps

**_Change the mail address to update the overview for the past commits_**

![sbmxg](https://user-images.githubusercontent.com/61122744/196498077-8a99551e-abd0-4b78-a8a0-1cb4373b4a71.png)

1. Open git bash
2. Clone a bare repository
`git clone --bare https://github.com/user/repo.git
cd repo.git`
3. Paste the following code into the git bash console after changing the variables `OLD_EMAIL,` `CORRECT_NAME `and `CORRECT_EMAIL`:
`#!/bin/sh

```
git filter-branch --env-filter '

OLD_EMAIL="your-old-email@example.com"
CORRECT_NAME="Your Correct Name"
CORRECT_EMAIL="your-correct-email@example.com"

if [ "$GIT_COMMITTER_EMAIL" = "$OLD_EMAIL" ]
then
export GIT_COMMITTER_NAME="$CORRECT_NAME"
export GIT_COMMITTER_EMAIL="$CORRECT_EMAIL"
fi
if [ "$GIT_AUTHOR_EMAIL" = "$OLD_EMAIL" ]
then
export GIT_AUTHOR_NAME="$CORRECT_NAME"
export GIT_AUTHOR_EMAIL="$CORRECT_EMAIL"
fi
' --tag-name-filter cat -- --branches --tags`
```
4. Press enter to run the script
5. Push the corrected history to github
`git push --force --tags origin 'refs/heads/*'`

![RLkHq](https://user-images.githubusercontent.com/61122744/196498171-6c639cd5-4f7d-4712-bbec-3a47d5f19893.png)

**Warning:** _This action is destructive to your repository's history. If you're collaborating on a repository with others, it's considered bad practice to rewrite published history. You should only do this in an emergency._

**Reply by henriquebaron** [reply]
Thank you for the reply. The issue was not in the e-mail address though, since my e-mail (and my profile picture) showed up correctly in the commit history.
I tracked the problem by looking at the commits in my local repo with `git log --pretty=fuller` and saw that the `AuthorDate` was wrong for some reason. So I had commits from which `AuthorDate` was more than a week older than the `CommitDate`. I found out that the contribution graph is based in the author date, and not the commit date.

What makes the author date in my commits differ totally from the date and time of my system is still unclear for me though. But I presume that's an issue with the virtual machine I'm working on.
